ksql-metastore/src/main/java/io/confluent/ksql/metastore/MetaStoreImpl.java

```diff
@@ -83,8 +83,18 @@ public StructuredDataSource getSource(final String sourceName) {
   public Optional<StructuredDataSource> getSourceForTopic(final String ksqlTopicName) {
     return dataSources.values()
         .stream()
-        .filter(p -> p.source.getKsqlTopic().getName() != null
-            && p.source.getKsqlTopic().getName().equals(ksqlTopicName))
+        .filter(p -> p.source.getTopicName() != null
+            && p.source.getTopicName().equals(ksqlTopicName))
+        .map(sourceInfo -> sourceInfo.source)
+        .findFirst();
+  }
+
+  @Override
+  public Optional<StructuredDataSource> getSourceForKafkaTopic(final String kafkaTopicName) {
+    return dataSources.values()
+        .stream()
+        .filter(p -> p.source.getKafkaTopicName() != null
```

Comment:
how is it possible for a `KsqlTopic` to have a `null` kafka topic name? Again out of scope for this PR, but unless I'm misunderstanding something we should probably push this null check into the constructor of `KsqlTopic`

Reply:
It shouldn't be possible, but this isn't the right PR for it. Will look into it more in a follow-up.